src/Hit/Issue.hs

```diff
@@ -164,3 +168,18 @@ parseOwnerRepo url =
 
     stripGitSuffix :: Text -> Maybe Text
     stripGitSuffix x = whenNothing (T.stripSuffix ".git" x) (Just x)
+
+fetchIssueStrict :: Id Issue -> IO Issue
+fetchIssueStrict iNum = fetchIssue iNum >>= \case
+    Left err -> errorMessage (show err) >> exitFailure
+    Right issue -> pure issue
+
+assignToIssue :: Issue -> IO (Either Error Issue)
+assignToIssue issue = withOwnerRepo action
+  where action Nothing _ _ = pure $ Left $ UserError "GITHUB_TOKEN lookup failed"
+        action (Just t) o r = do
+          let currentAssignees = simpleUserLogin <$> issueAssignees issue
+              newAssignee = makeName $ untagName o
+              newAssignees = V.cons newAssignee currentAssignees
+              isn = mkIssueId $ unIssueNumber $ issueNumber $ issue
+          editIssue t o r isn editOfIssue {editIssueAssignees = Just newAssignees}
```

Comment:
My feeling is that the function shouldn't do any requests if you're already assigned to an issue

Reply:
My new [assignAction](https://github.com/StevenXL/hit-on/blob/a2fe9f891998d92cdfe090be06d73114f07e1197/src/Hit/Issue.hs#L174) seems very verbose to me. Any way to clean it up while still being able to discriminate between the case where the user was already assigned to the issue versus the case where they are newly assigned to the issue?